pytools/graph.py

```diff
@@ -369,4 +370,40 @@ def compute_induced_subgraph(graph: Mapping[T, Set[T]],
 
 # }}}
 
+
+# {{{
+
+def visualize_graph(graph: Mapping[T, Iterable[T]]) -> None:
+    """
+    Visualize the graph *graph*. Needs the ``dot`` program installed.
+
+    :arg graph: A :class:`collections.abc.Mapping` representing a directed
+        graph. The dictionary contains one key representing each node in the
+        graph, and this key maps to a :class:`collections.abc.Set` of nodes
+        that are connected to the node by outgoing edges.
+    """
+    from pytools import UniqueNameGenerator
+    id_gen = UniqueNameGenerator(forced_prefix="mynode")
+
+    from pytools.dot import dot_escape, show_dot
+
+    array_to_id = {}
+
+    res = "digraph mygraph {\n"
```

Comment:
Use f-strings better here, maybe

```python
result = "digraph mygraph {{\n{ content }\n}}"
```
So that open/close brace remain close in code.

Reply:
Should be fixed in 
4dd3b81